aspnetcore/blazor/includes/js-interop/circuit-disconnection.md

```diff
@@ -0,0 +1,46 @@
+*This section only applies to Blazor Server apps.*
+
+JavaScript (JS) interop calls can't be issued after a SignalR circuit is disconnected. Without a circuit during component disposal or at any other time that a circuit doesn't exist, the following method calls fail and log a message that the circuit is disconnected as a <xref:Microsoft.JSInterop.JSDisconnectedException>:
+
+* JS interop method calls
+  * <xref:Microsoft.JSInterop.IJSRuntime.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync%2A?displayProperty=nameWithType>)
+* `Dispose`/`DisposeAsync` calls on any <xref:Microsoft.JSInterop.IJSObjectReference> or `Dispose` calls on any <xref:Microsoft.JSInterop.DotNetObjectReference>.
+
+In order to avoid logging <xref:Microsoft.JSInterop.JSDisconnectedException> or to log custom information, catch the exception in a [`try-catch`](/dotnet/csharp/language-reference/keywords/try-catch) statement.
+
+For the following component disposal example with an <xref:Microsoft.JSInterop.IJSObjectReference> object used to call JS from .NET:
+
+* The component implements <xref:System.IAsyncDisposable>.
+* `objInstance` is an <xref:Microsoft.JSInterop.IJSObjectReference>.
+* <xref:Microsoft.JSInterop.JSDisconnectedException> is caught and not logged.
+* Optionally, you can log custom information in the `catch` statement at whatever log level you prefer. The following example doesn't log custom information because it assumes the developer doesn't care about when or where circuits are disconnected during component disposal.
+
+```csharp
+async ValueTask IAsyncDisposable.DisposeAsync()
+{
+    try
+    {
+        if (objInstance is not null)
+        {
+            await objInstance.DisposeAsync();
+        }
+    }
+    catch (JSDisconnectedException)
+    {
+    }
+}
+```
+
+> [!NOTE]
+> The pattern for <xref:Microsoft.JSInterop.DotNetObjectReference> disposal is the same in scenarios where .NET is called from JS.
+>
+> When a circuit fails and developer disposal code fails with a caught or uncaught <xref:Microsoft.JSInterop.JSDisconnectedException>, automatic garbage collection eventually disposes of any objects created by the developer in the component.
+
+If you must clean up your own JS objects or execute other JS code on the client after a circuit is lost or you prefer to dispose of one or more <xref:Microsoft.JSInterop.IJSObjectReference>s or <xref:Microsoft.JSInterop.DotNetObjectReference>s on the client, use the [`MutationObserver`](https://developer.mozilla.org/docs/Web/API/MutationObserver) pattern in JS on the client and avoid calling `Dispose`/`DisposeAsync` on the objects in the component.
```

Comment:
> and avoid calling `Dispose`/`DisposeAsync` on the objects in the component.

I think that clause could be dropped, as it should be safe to request disposal from the server whether or not the client is also disposing the same object.

Reply:
Safe, yes ... but that misses the answer to @Liero's ask. The ask is like, 'How do I avoid the logging/`JSDisconnectedException`?' Javier said to either ...

* Catch it and do nothing ... or ...
* Use `MutationObserver` as an alternative to disposing in C# to avoid the exception completely.